src/test/test-unit-name.c

```diff
@@ -106,14 +106,21 @@ TEST(unit_name_replace_instance) {
 
 static void test_unit_name_from_path_one(const char *path, const char *suffix, const char *expected, int ret) {
         _cleanup_free_ char *t = NULL;
+        int r;
 
         assert_se(unit_name_from_path(path, suffix, &t) == ret);
         puts(strna(t));
         assert_se(streq_ptr(t, expected));
 
         if (t) {
                 _cleanup_free_ char *k = NULL;
-                assert_se(unit_name_to_path(t, &k) == 0);
+
+                /* Unit names that contain "..." in the prefix are created from paths that would otherwise produce unit names that are too long. */
+                r = unit_name_to_path(t, &k);
+                assert_se(r == 0 || r == -ENAMETOOLONG);
```

Comment:
move this assert_se() line after the if check below, and you don't have to check for ENAMETOOLONG twice